src/main/java/spoon/support/compiler/jdt/JDTTreeBuilder.java

```diff
@@ -342,6 +342,9 @@ public void endVisit(Block block, BlockScope scope) {
 
 	@Override
 	public void endVisit(BreakStatement breakStatement, BlockScope scope) {
+		if (breakStatement.isSynthetic) {
+			return; // we never entered
```

Comment:
Why handle this then instead of throwing an exception?

Reply:
This method is called by JDT, whether we want or not (i.e. even if we return false in visit(BreakStatement, ...))

Reply:
ah right we would need to return false in the *parent* construct. That makes sense